File: seed/csharp-sdk/trace/src/SeedTrace/V2/Problem/Types/GetBasicSolutionFileRequest.cs
using System.Text.Json.Serialization;
using SeedTrace.Core;

#nullable enable

namespace SeedTrace.V2;

public record GetBasicSolutionFileRequest
{
    [JsonPropertyName("methodName")]
    public required string MethodName { get; set; }

    [JsonPropertyName("signature")]
    public required NonVoidFunctionSignature Signature { get; set; }

    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: seed/php-model/trace/src/V2/Problem/GetBasicSolutionFileRequest.php
<?php

namespace Seed\V2\Problem;

use Seed\Core\SerializableType;
use Seed\Core\JsonProperty;
use Seed\V2\Problem\NonVoidFunctionSignature;

class GetBasicSolutionFileRequest extends SerializableType
{
    #[JsonProperty("methodName")]
    /**
     * @var string $methodName
     */
    public string $methodName;

    #[JsonProperty("signature")]
    /**
     * @var NonVoidFunctionSignature $signature
     */
    public NonVoidFunctionSignature $signature;

    /**
     * @param string $methodName
     * @param NonVoidFunctionSignature $signature
     */
    public function __construct(
        string $methodName,
        NonVoidFunctionSignature $signature,
    ) {
        $this->methodName = $methodName;
        $this->signature = $signature;
    }
}
